lib/kennel/api.rb

```diff
@@ -145,5 +145,23 @@ def with_cache(enabled, key)
         result
       end
     end
+
+    def tag(api_resource, reply)
+      ResourceTaggedHash.new(api_resource, reply)
+    end
+
+    class ResourceTaggedHash < Hash
```

Comment:
careful with subclassing hash it can get weird ... for example after merging methods might be gone etc
also check performance in case things get slower when not using native hashes